src/sagemaker_sklearn_extension/preprocessing/encoders.py

```diff
@@ -620,3 +621,248 @@ def inverse_transform(self, X):
+    not affect the performance of any supervised model applied thereafter.
+    See [1] for more details on WoE.
+
+    Parameters
+    ----------
+    feature_indices: list of integer indices (default=None)
+        Indices of the features to encode with WoE. If set to `None`, all
+        features will be encoded.
+
+    binning: {'uniform', 'quantile', 'kmeans', None}, default=None
+        What binning method to apply if not set to None. See [2].
+
+    n_bins: int (default=10), greater than 2
+        Number of bins to use if binning is to be applied.
+
+    alpha: float (default = 0.5), non-negative
+        Regularization value to avoid numerical errors due to division by
+        zero in the computation of the weight of evidence (e.g. in case the
+        data points corresponding to one category of a feature all have
+        the same target value).
+
+    laplace: boolean (default = False)
+        If alpha is positive, adds Laplace smoothing to the computation
+        of the weight of evidence.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> from woe import WOEEncoder
+    >>> sex = np.random.choice(['m', 'f'], size=100)
+    >>> age = np.random.randint(low=25, high=95, size=100)
+    >>> y = np.random.choice([0, 1], size=100)
+    >>> e_sex = WOEEncoder().fit_transform(sex, y)
+    >>> e_age = WOEEncoder(binning='quantile', n_bins=5).fit_transform(age, y)
+
+    Attributes
+    ----------
+    _beta: float
+        Value used in the Laplace smoothing when smoothing is used. See [3].
+
+    _count_y: list of size 2
+        Number of observations for each of the two target classes.
+
+    _mask_y_0: array (n_samples,)
+        Mask of observations for which the target is the first one.
+
+    _mask_y_1: array (n_samples,)
+        Mask of observations for which the target is the second one.
+
+    _binners: list of size n_encoded_features
+        List of fitted binning encoders.
+
+    _woe_pairs: list of pairs (codex, woe) of size n_encoded_features
+        The codex has the mapping feature_value => woe_index and woe has the
+        weight of evidence values.
+
+    References
+    ----------
+    [1] https://www.listendata.com/2015/03/weight-of-evidence-woe-and-information.html
+    [2] https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.KBinsDiscretizer.html
+    [3] https://en.wikipedia.org/wiki/Additive_smoothing
+    """
+
+    def __init__(self, feature_indices=None, binning=None,
+                 n_bins=10, alpha=0.5, laplace=False):
+        self.feature_indices = feature_indices
+        self.binning = binning
+        self.n_bins = n_bins
+        self.alpha = alpha
+        self.laplace = laplace
+
+        # fail early for arguments that don't meet the guidelines
+        if binning:
+            assert binning in ("uniform", "quantile", "kmeans"), \
+                WOEAsserts.BINNING
+            assert n_bins > 2, WOEAsserts.NBINS
+        assert alpha >= 0, WOEAsserts.ALPHA
```

Comment:
I believe this will fail `check_estimator`, there's a requirement that all input validation be done in `fit` rather than the initialization.

Reply:
I'll try that